### PR TITLE
Add control over associations params

### LINKED ISSUE #15: Nested params
Currently, we had an effect of _params shadowing_: nested cloner could rely on the same params as the top-level ones, but they must not know about each other (more precisely, children cloners must not know about parent cloners).

We need a way to control how `params` are passed to associations cloners, e.g.:

```ruby
# just pass params as is (current behaviour)
include_association :users, params: true

# use custom block to prepare params
include_association :users, params: ->(params) { params[:data] }

# or use a key
include_association :profile, params: :profile

# usage
UserCloner.call(user, profile: { reset: true })
```

I also propose to **not proxy params** by default to the underlying cloners.

### PR DESCRIPTION
**Pull request by ssnickolay**
Closes #15
- [x] Implement new nested parameters logic.
- [x] Change integration specs with using new logic.
- [x] Add new documentation page about parameters.
- [x] Сheck for correctness all docs snippets.
- [x] ~Support multiple keys.~
- [x] Pass the parent record as a param.

### CODE REVIEW

**Comment by ssnickolay** [comment]
>  let's rename ByBlockFilter and ByKeyFilter to just BlockFilter and KeyFilter respectively.

I agree :+1:

> And for AllowFilter and DenyFilter, maybe, it would be better to use something like PassFilter and NullFilter.

I think `NullFilter` not very good  class name, because it's sounds like `EmptyFilter` or `WithoutFilter`(

**Comment by palkan** [comment]
> I think NullFilter not very good class name, because it's sounds like EmptyFilter or WithoutFilter(

What about using `Proxy` instead of `Filter`? Then `NullProxy` makes sense

**Comment by ssnickolay** [comment]
> What about using Proxy instead of Filter? Then NullProxy makes sense

Hm, I like it :+1:

File: docs/parameters.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  trait :params_true do
+    include_association :profile, params: true
+  end
+
+  trait :by_key do
+    include_association :profile, params: :profile
+  end
+
+  trait :by_block do
+    include_association :profile, ->(params) {
```
Comment (palkan) [inline docs/parameters.md]: `params: -> (params) { ... }` ?

Comment (ssnickolay) [reply docs/parameters.md]: Damn, thx!

**Comment by ssnickolay** [comment]
@palkan It looks like we're done. Can you look again?

File: docs/parameters.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  # Filter parameters by key.
+  # Notice: value by key must be a Hash.
+
+  trait :by_key do
+    include_association :profile, params: :profile
+  end
+
+  # Execute custom block with params as argument
+  trait :by_block do
+    include_association :profile, params: lambda do |params|
```
Comment (palkan) [inline docs/parameters.md]: Maybe it's better to encourage users to use `proc`?

**Comment by ssnickolay** [comment]
@palkan done :+1: Feel free to merge to master)

**Pull request merged**